actions/snapshot

```diff
@@ -24,11 +24,13 @@ DATE_STAMP=$(date +%Y-%m-%d-%H.%M.%S)
 ARCHIVE=etcd-snapshot-$DATE_STAMP.tar.gz
 
 # Ensure the backupd target exists
-mkdir -p $ETCD_BACKUP_TARGET_DIR/$JUJU_ACTION_UUID
+mkdir -p $ETCD_BACKUP_TARGET_DIR/
 
 if [ "${ETCD_KEYS_VERSION}" == "v2" ]; then
-  # Dump the data currently in the cluster
-  ETCDCTL_API=2 /snap/bin/etcd.etcdctl backup --data-dir $ETCD_DATA_DIR --backup-dir $ETCD_BACKUP_TARGET_DIR/$JUJU_ACTION_UUID
+  # Dump the data currently in the cluster from the snap, then move to $ETCD_BACKUP_TARGET_DIR
+  V2_SNAPSHOT=$ETCD_DATA_DIR/v2-snapshots/$JUJU_ACTION_UUID
+  ETCDCTL_API=2 /snap/bin/etcd.etcdctl backup --data-dir $ETCD_DATA_DIR --backup-dir $V2_SNAPSHOT
+  mv $V2_SNAPSHOT $ETCD_BACKUP_TARGET_DIR/
 elif [ "${ETCD_KEYS_VERSION}" == "v3" ]; then
   mkdir -p $ETCD_BACKUP_TARGET_DIR/$JUJU_ACTION_UUID
   cp $ETCD_DATA_DIR/member/snap/db $ETCD_BACKUP_TARGET_DIR/$JUJU_ACTION_UUID
```

Comment:
careful `bash` adjustment here to make up for the fact that in jammy, snapd confinement is more serious about not letting `etcdctl` write to just anywhere.  Now for v2 snapshots, this script will write to the `/var/snap/etcd/v2-snapshots/` directory, then move that backup to the target directory after the fact.